contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,568 @@
+name: myclaim
+spec:
+  volumeMode: Block #proposed API change
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 80Gi 
+```
+
+For dynamic provisioning and the use of the storageClass, the admin also specifically defines the intent of the volume by 
+indicating the volumeMode as block. The provisioner for this class will validate whether or not it supports block and return
+an error if it does not.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+name: myclaim
+spec:
+  storageClassName: local-fast 
+  volumeMode: Block #proposed API change
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 80Gi 
+```
+
+## Persistent Volume API Changes:
+For static provisioning the admin creates the volume and also is intentional about how the volume should be consumed. For backwards
+compatibility, the absence of volumeMode will default to file which is how volumes work today, which are formatted with a filesystem depending on the plug-in chosen. Recycling will not be a supported reclaim policy as it has been deprecated. The path value in the local PV definition would be overloaded to define the path of the raw block device rather than the fileystem path.
+```
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+name: local-raw-pv
+spec:
+  volumeMode: Block #proposed API change
+  capacity:
+    storage: 100Gi
+  local:
+    path: /dev/xvdc #device path
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Delete 
+```
+## Pod API Changes:
+To provide better specificy and ensure support of inline volumes, the following changes are proposed in the pod specification.
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-db
+spec:
+    containers
+    - name: mysql
+      image: mysql
+      volumeDevices: #proposed API change
+      - name: my-db-data
+        devicePath: /dev/xvda #proposed API change
+    volumes:
+    - name: my-db-data
+      persistentVolumeClaim:
+	claimName: raw-pvc
+```
+## Storage Class non-API Changes:
+For dynamic provisioning, it is assumed that values pass in the parameter section are opaque, thus the introduction of utilizing
+fsType in the StorageClass can be used by the provisioner to indicate how to create the volume. The proposal for this value is
+defined here:
+https://github.com/kubernetes/kubernetes/pull/45345 
+Therefore, a provisioner could potentially provision a block device and install the filesystem onto it by indicating the volumeMode
+as 'block' but the fsType as 'xfs'.
+This section is provided as a general guideline, but each provisioner may implement their parameters independent of what is defined
+here. It is our recommendation that the volumeMode in the PVC be the guidance for the provisioner and overrides the value given in the fstype. Therefore a provisioner should be able to ignore the fstype and provision a block device if that is what the user requested via the PVC and the provisioner can support this.
```

Comment:
volumeMode must be the guidance.  We will not bind a PVC.volumeMode == filesystem if the provisioner created a PV.volumeMode == block, and vice versa.